src/main/java/com/aws/greengrass/componentmanager/builtins/GreengrassRepositoryDownloader.java

```diff
@@ -234,7 +234,7 @@ private String getArtifactDownloadURL(ComponentIdentifier componentIdentifier, S
     }
 
     SdkHttpClient getSdkHttpClient() {
```

Comment:
this is still returning a _built_ client, not a builder. So is the user of `getSdkHttpClient` handling that fact correctly?

Reply:
All users of `getSdkHttpClient` call it in a try with resource directly and not in a SDK builder. I verified that `client.close()` is called after try with resource block finishes.